zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/ElasticsearchEngineClient.java

```diff
@@ -258,25 +259,27 @@ private PutIndexTemplateRequest putIndexTemplateRequest(
       final IndexSettings settings,
       final Boolean create) {
 
-    try (final var templateMappings =
+    try (final var templateFile =
         getResourceAsStream(indexTemplateDescriptor.getMappingsClasspathFilename())) {
 
+      final var templateFields = deserializeJson(IndexTemplateMapping._DESERIALIZER, templateFile);
+      final var templateSettings = templateFields.settings();
+
       return new PutIndexTemplateRequest.Builder()
           .name(indexTemplateDescriptor.getTemplateName())
           .indexPatterns(indexTemplateDescriptor.getIndexPattern())
           .template(
               t ->
                   t.aliases(indexTemplateDescriptor.getAlias(), Alias.of(a -> a))
-                      .mappings(
-                          deserializeJson(IndexTemplateSummary._DESERIALIZER, templateMappings)
-                              .mappings())
+                      .mappings(templateFields.mappings())
                       .settings(
```

Comment:
@EuroLew Did I get it right, that we can't simply use `.settings(templateSettings)` here because we need to supplement settings read from file with number of shards and number of replicas?
I'm worried here, that we read settings from file as they are but take only couple of properties from them (analisys and index). Could be error prone in the future if we decide to use other setting "types" also. Do you think we can somehow use the full settings object from the file? 🤔

Reply:
yes we can't use .settings due to the need to supplement, and yes you are correct it is error prone in the future as it will not pick up other settings types. The solutions are 
- capture every settings type this would be like a dozen+ lines? I could add this just not pretty
- add an endpoint to specifically update the settings of templates, I don't like this solution because this is what PUT index templates is meant to support but that currently derives from the .json file so it would require deserialisation into a json type/map and edit that map then serialise back in an input stream for the request

I

Reply:
I resolved it by deserialising, adding the desired fields, serialising back, its a costly operate but we won't need to do it often (only on creation) so should be fine.

Reply:
I would still try to manipulate Json in some way. Like reading `settings` part to map, putting number of shards/replicas and then somehow convert it to `IndexSettings` object. It will be ugly probably, but should be safe. What do you think?

If not, I'm fine with the 1st solution.